test/functional/test_requires.py

```diff
@@ -0,0 +1,77 @@
+from collections import namedtuple
+
+import pytest
+
+from taskotron_python_versions.requires import (
+    get_versioned_name,
+    check_requires_naming_scheme,
+)
+
+from .common import gpkg
+
+
+Package = namedtuple('Pcakage', 'name')
+
+
+class QueryStub(object):
+
+    """Stub object for dnf repoquery."""
+
+    def __init__(self, data):
+        """Provide data as {<provide_name>: [<package_name>, ..], ..}, where
+            `provide_name`: (str) name expected to be passed to filter method
+            `package_name`: (str) package name providing it in query result
+        """
+        self.data = data
+        self.provides = None
+
+    def filter(self, provides):
+        self.provides = provides
+        return self
+
+    def run(self):
+        if self.provides not in self.data:
+            raise Exception('Running repoquery on {} is not expected'.format(
```

Comment:
Is the general Exception here on purpose or just for the lack of a more specific one? I'd go with ValueError here I guess.